scripts/pre_commit_linter.py

```diff
@@ -1000,17 +1000,18 @@ def _validate_and_parse_js_and_ts_files(self):
                 # file.
                 if filepath.endswith('.js'):
                     raise Exception(e)
-                compiled_js_filepath = self._compile_ts_file(filepath)
+                compiled_js_filepath = self._compile_ts_file(
+                    filepath, compiled_js_dir)
                 file_content = FileCache.read(compiled_js_filepath).decode(
                     'utf-8')
                 parsed_js_and_ts_files[filepath] = esprima.parseScript(
                     file_content)
 
-        shutil.rmtree(self.compiled_js_dir)
+        shutil.rmtree(compiled_js_dir)
 
         return parsed_js_and_ts_files
 
-    def _compile_ts_file(self, filepath):
+    def _compile_ts_file(self, filepath, compiled_js_dir):
```

Comment:
- Should we make this a static method? (not sure!)
- for this method `compiled_js_dir` is a general directory, right? If yes then maybe we can have a general parameter name.

Reply:
I am sorry but I did not get this one. What do you mean by general directory here?

Reply:
> for this method compiled_js_dir is a general directory,

`_compile_ts_file` method needs two parameters:
1. Path of the .ts file.
2. Path to a dir. in which we the .js file will be populated.

So can we rename the 2nd parameter to `dir_path` or something around this? (we have a similar name for the first parameter.)

Reply:
Ah, got it. Updated now.